src/OrchardCore.Modules/OrchardCore.Media/Startup.cs

```diff
@@ -190,22 +166,30 @@ public override void ConfigureServices(IServiceCollection services)
         public override void Configure(IApplicationBuilder app, IRouteBuilder routes, IServiceProvider serviceProvider)
         {
             var mediaFileProvider = serviceProvider.GetRequiredService<IMediaFileProvider>();
+            var mediaOptions = serviceProvider.GetRequiredService<IOptions<MediaOptions>>();
+            var mediaFileStoreCacheProvider = serviceProvider.GetService<IMediaFileStoreCacheProvider>();
+
+            // FileStore middleware before ImageSharp, but only if a remote storage module has registered a cache provider.
+            if (mediaFileStoreCacheProvider != null)
+            {
+                app.UseMiddleware<MediaFileStoreResolverMiddleware>();
```

Comment:
Middleware only gets used if a file store cache provider is registered, by another module (blob). So doesn't need to no-op when using standard media folder, just isn't in the pipeline at all.